storage/innobase/xtrabackup/src/backup_mysql.cc

```diff
@@ -1577,6 +1592,9 @@ write_binlog_info(MYSQL *connection)
 	gtid = (gtid_executed != NULL ? gtid_executed : gtid_current_pos);
 
 	if (mariadb_gtid || mysql_gtid) {
+		if (mysql_gtid && mysql_server_version >= 50705) {
+			flush_binlogs_maybe(connection);
```

Comment:
This is a fair point and can also apply to the Galera/SST case.   Here, the flush is not strictly required but rather a convenience to avoid a case where the `mysql.gtid_executed` table is inconsistent after restore.

As far as performance impact, it is not hard to contrive a case with a large amount of "expired" binlogs that are purged on flush that can take many seconds or even minutes in extreme cases.   Based on that alone, I understand if this PR is rejected.